app/services/random_phrase.rb

```diff
@@ -16,7 +16,10 @@ def to_s
 
   def build_words(num_words)
     str_size = num_words * @word_length
-    random_string = SecureRandom.base64(str_size).gsub(%r{[0o+\/=]}, ('A'..'Z').to_a.sample)
-    random_string.upcase[0..(str_size - 1)].chars.each_slice(@word_length).map(&:join)
+    # 5 bits per character means we must multiply what we want by 5
+    # :length adds zero padding in case it's a smaller number
+    random_bytes = SecureRandom.random_number(2**(str_size * 5))
+    random_string = Base32::Crockford.encode(random_bytes, length: 16)
+    random_string.upcase.chars.last(str_size).each_slice(@word_length).map(&:join)
```

Comment:
The `upcase` is definitely a belt-and-suspenders redundancy because I wanted to be sure we were handing back UPPER case strings No Matter What.

The `last(str_size)` will likely be resolved if we set `length: str_size` instead.